src/main/java/org/jabref/logic/importer/fetcher/JstorFetcher.java

```diff
@@ -82,6 +84,46 @@ public URL getURLForQuery(ComplexSearchQuery query) throws URISyntaxException, M
         return uriBuilder.build().toURL();
     }
 
+    @Override
+    public URL getUrlForIdentifier(String identifier) throws FetcherException {
+        String start = "https://www.jstor.org/citation/text/";
+        if (identifier.startsWith("http")) {
+            identifier = identifier.replace("https://www.jstor.org/stable", "");
+            identifier = identifier.replace("http://www.jstor.org/stable", "");
+        }
+        identifier = identifier.replaceAll(URL_QUERY_REGEX, "");
+
+        try {
+            if (identifier.contains("/")) {
+                // if identifier links to a entry with a valid doi
+                return new URL(start + identifier);
+            }
+            // else use default doi start.
+            return new URL(start + "10.2307/" + identifier);
+        } catch (IOException e) {
+            throw new FetcherException("could not construct url for jstor", e);
+        }
+    }
+
+    // overwriting default behaviour as that seems to always results in a 403
```

Comment:
What did you had to change relative to the standard implementation?

Reply:
getParser was already written to only work for search so I simply overwrote this method to use a bibtex parser.
I did have another look at the code, removed this part and added the required logic to getParser.